crates/hdx_parser/src/macros.rs

```diff
@@ -282,3 +282,17 @@ macro_rules! match_ignore_case {
 		}
 	};
 }
+
+#[macro_export]
+macro_rules! match_delim {
+       ( $parser: ident.$method: ident($($args: tt)*):
+           $(
+              $pattern:pat $(if $guard:expr)?  => $then: expr
+           ),+
+           $(,)?
+       ) => {
+		match $parser.$method($($args)*).char() {
+		    $(Some($pattern) $( if $guard )? => $then,)+
+		}
+	};
+}
```

Comment:
@keithamus: something like this? I'm noticing now that I don't have the `.kind()` check in here now. I still need that right? Should I do the match here for a tuple like I was doing earlier and always make sure that `.kind() == Kind::Delim`?

Reply:
Ideally it should check `.kind()` because `.char()` returns values for the other single character tokens like `RightParen`. In this particular case it's okay because we have a fallthrough to `unexpected!` but we should refine this later.